backend/fms_core/viewsets/study.py

```diff
@@ -55,4 +58,19 @@ def create(self, request, *args, **kwargs):
         else: 
             # Serialize study
             serializer = StudySerializer(study)
-            return Response(serializer.data)
\ No newline at end of file
+            return Response(serializer.data)
+
+    def destroy(self, request, pk=None):
+        errors = defaultdict(list)
+
+        if pk is None:
+            errors['Study'].append("pk cannot be None")
+        else:
+            _, newerrors, _ = can_remove_study(pk)
+            errors = merge_dicts_of_list(newerrors, errors)
```

Comment:
This is a very busy way of doing this. It is quite generic but i think a bit overkill i would suggest you pass a list from errors and limit you service to returning a single error list (as is done elsewhere in the application).  For example, removable, errors['Removable'], _ = can_remove_study(pk).